src/renderer/base/RenderEngineBase.cpp

```diff
@@ -7,38 +7,25 @@
 using namespace Microsoft::Console;
 using namespace Microsoft::Console::Render;
 
-RenderEngineBase::RenderEngineBase() :
-    _titleChanged(false),
-    _lastFrameTitle(L"")
+HRESULT RenderEngineBase::InvalidateTitle() noexcept
 {
-}
-
-HRESULT RenderEngineBase::InvalidateTitle(const std::wstring_view proposedTitle) noexcept
-{
-    if (proposedTitle != _lastFrameTitle)
-    {
-        _titleChanged = true;
-    }
-
+    _titleChanged = true;

```

Comment:
Do we actually need to ensure we only trigger a title change if the title _actually_ changed?
Personally I don't think so so I removed the logic for it.

Reply:
I think this eventually leads to some event in conhost? I don't quite remember. 

If there's some application that rapidly sends `titleChange` event with the same title, it may exhausts the event handler. So I actually left this logic in my PR.

Reply:
But is there any serious application which rapidly updates the title with the same value again and again?

Reply:
I'm more concerned about the misbehaving ones :smile:

Reply:
To be fair, such an application could simply misbehave by setting different titles on purpose.

I wonder if there's an application in common use that keeps re-setting the same title. Maybe @miniksa knows why this check exists.

It seems unlikely that it's just to save time on a relatively uncommon operation.

Reply:
the check exists because:
* The VT renderer doesn't want to send an updated title if it's the same. That's more noise than it needs, we're trying to be a little less chatty.
* Conhost also updates the title by sending a custom window message to itself, which is a little weird but hey that's how it does it. But there definitely was a misbehaving app or scenario or something in the past that would cause the title to change way more frequently than 60hz, so we added this check. It was definitely a hot path, cause it had to go out through User32<sup>(?)</sup> and back. Over the life of conhost, that check eventually got migrated to the GdiEngine, and at that point, both heads were doing the check, so we figured "eh, might as well have a render engine base class that does things like this, that won't change between engines ¯\\\_(ツ)_/¯"